File: python/day01a.py
import math


def main():
    with open("input/day01.txt") as file:
        result = 0
        last = math.inf
        for line in file.readlines():
            current = int(line)
            if current > last:
                result += 1
            last = current
        print(f"{result} measurements are larger")


if __name__ == "__main__":
    main()

File: src/bin/day01a.rs
use std::fs::File;
use std::io::{BufRead, BufReader};

type Error = Box<dyn std::error::Error>;

fn main() -> Result<(), Error> {
    let file = File::open("input/day01.txt")?;
    let reader = BufReader::new(file);
    let mut last = i32::MAX;
    let mut result = 0;
    for line in reader.lines() {
        let current = line?.parse::<i32>()?;
        if current > last {
            result += 1;
        }
        last = current;
    }
    println!("{} measurements are larger", result);
    Ok(())
}
